### DISCUSSION TITLE
Unit Test By Passing Header

### DISCUSSION THREAD

**Discussion by dpatra1** [Netflix/dgs-framework#353] [Q&A]
Hi,

I have a bunch of graphql queries which requires to have few headers. With the current version of DGS, I can not use `executeAndExtractJsonPath()` method of `DgsQueryExecutor` because this does not accept a header object. 

So I was trying to use `fun execute(query: String, variables: Map<String, Any>, extensions: Map<String, Any>?, headers: HttpHeaders)` function in my test but looks like its not triggering my datafetcher only. I could not figure out what is the mistake I am doing here.

Here is my code:

    HttpHeaders httpHeaders = new HttpHeaders();   
    httpHeaders.set(HEADER_NAME, value);

    when(service.find(eq(id)))
        .thenReturn(response);

    Map<String, Object> variables = new HashMap<>();
    variables.put(ID, id);

    ExecutionResult response =
        dgsQueryExecutor.execute(GET_MAPPER_BY_SHEERID_SCHOOL_ID, variables, null, httpHeaders);

    Assertions.assertNotNull(response);

Any help would be appreciated!

Thanks In advance

**Comment by sbilello** [comment]
Try to look at this PR https://github.com/Netflix/dgs-framework/pull/302/files

**Reply by dpatra1** [reply]
@sbilello Looks good to me. Is this already released ?

**Reply by sbilello** [reply]
Yes, I am not asking for a code review. Maybe that code example is what you are looking for.

**Reply by dpatra1** [reply]
@sbilello Which version includes these changes? Because I have graphql-dgs-3.10.2 and I dont see this method in DgsQueryExecutor.

**Comment by sbilello** [comment]
Did you try the latest version?

**Reply by dpatra1** [reply]
@sbilello Looks like the version is hard coded in our internal start dependency jar due to which I am not getting the latest version.
`implementation platform("com.netflix.graphql.dgs:graphql-dgs-platform-dependencies:${netflixDgsVersion}")`

Is there a way we can automatically pull the latest release version when you are releasing from your end instead of we configure the version in our end ?

it will be really helpful if you can help me  on that. I heard from one of my team member that you guys have way to enforce the platform now.

Thanks,
Deba

**Comment by sbilello** [comment]
What you are asking looks like something related to
https://github.com/dependabot CI/CD pipeline in the automation.
I don't work for Netflix.
Best,
Sergio

**Reply by dpatra1** [reply]
@sbilello I have tried with the latest version of DGS, but problem still persists. I still see the header is coming as null.

Does anyone has faced this similar issue? Do we have a solution for this?

DataFetcher
`  @DgsQuery
  public CustomResult pullData(
      @InputArgument CustomRequest req,
      @RequestHeader("CLIENT_ID") String clientId) {
    return new CustomResult(
        service.pullData(req, clientId));
  }`

UnitTest For DateFetcher
`  public static final String QUERY =
      "{\n"
          + "  pulldata(hsExamPreSignedUrlReq: {fileName : \"ABC.pdf\"}) {\n"
          + "    customResultMetadata {\n"
          + "      name\n"
          + "    }\n"
          + "  }\n"
          + "}";

  @Test
  void pullDataTest() {
    MultiValueMap<String, String> map = new LinkedMultiValueMap<>();
    map.add(CLIENT_ID, UUID.randomUUID().toString());
    HttpHeaders httpHeaders = new HttpHeaders(map);

    CustomResult response = new CustomResult(clientId, name);

    String value =
        dgsQueryExecutor.executeAndExtractJsonPath(
            QUERY,
            "data.pulldata.customResultMetadata.name",
            httpHeaders);
            
    assertThat(value).isEqualTo("test");
  }`
  
  
  When the test is getting executed, `@RequestHeader("CLIENT_ID") String clientId` value is going as null.
  
  Thanks In Advance
  Deba

**Reply by srinivasankavitha** [reply]
Could you please try this set up as a workaround while we fix the original API:

```
val request = MockHttpServletRequest().addHeader("foo", "bar")
executor.execute(
      query,
       variables,
       emptyMap(),
       HttpHeaders.EMPTY,
       null,
       ServletWebRequest(request)
)
```

**Reply by dpatra1** [reply]
@srinivasankavitha I have tried the work around but I still see the null value when I am trying to read the header in DataFetcher. I would wait for your fix on this and test it after that.

Thanks for looking into this.

Thanks,
Deba

**Reply by brianlschultz1969** [reply]
Any solution for this issue yet? I am facing same - we are on version 8.5.8
